### DISCUSSION TITLE
Telescope Lens Diameter doesn't exist

### DISCUSSION THREAD

**Discussion by TheOnlyMrVideo** [Stellarium/stellarium#3298] [General]
Stellarium tends to be geared toward the use of a telescope, even though you can define a telescope with a camera lens, I'm assuming to fake it out as a camera. When defining said lens, there is a Diameter field to be filled in. Problem is that a camera lens doesn't have a diameter aperture value. At least one that I've never used or seen in the specs. The closest thing I can think of is the f-stop, which doesn't into the integer field. So, what does one enter there? The manual provides no help in this case.

The provided examples have values in the Diameter field. I have no clue where they came from.

**Comment by alex-w** [comment]
Hi!

пн, 3 июл. 2023 г. в 11:59, TheOnlyMrVideo ***@***.***>:

> Stellarium tends to be geared toward the use of a telescope, even though
> you can define a telescope with a camera lens, I'm assuming to fake it out
> as a camera. When defining said lens, there is a Diameter field to be
> filled in. Problem is that a camera lens doesn't have a diameter aperture
> value. At least one that I've never used or seen in the specs. The closest
> thing I can think of is the f-stop, which doesn't into the integer field.
> So, what does one enter there? The manual provides no help in this case.
>
It's maybe strange, but can you share examples for camera lenses?

-- 
With best regards, Alexander

**Comment by 10110111** [comment]
Generally, this field doesn't affect any geometry. It's only used to calculate limiting magnitude and to show some textual info. So you could just copy-paste this value from another lens entry.

But, if you really need it, I suppose it's the same as aperture size for a photo lens, so given the f-number and the focal distance of your lens, you can just use the formula *D=f/N*, where *f* is focal distance, and *N* is the f-number.

**Reply by TheOnlyMrVideo** [reply]
OK, thanks.

**Comment by gzotti** [comment]
BTW, telescope support and ocular/sensor simulation is done as optional plugins, so it's not a primary feature, and there are applications of astronomy research where these plugins are of no use. Unaided observing, placing yourself in the field is core.

**Reply by TheOnlyMrVideo** [reply]
I don't use the program as a main source for photography. The main use is to show me what the sky will be at a particular time and place. The ability to see what the camera lens will see is a bonus.
